### PR TITLE
Add previous step name to back button for flow navigation

### PR DESCRIPTION
**Pull request by oliverviljamaa**
Issue: https://transferwise.atlassian.net/browse/CON-1068

# v3.1.0
## Add previous step name to back button for flow navigation

flow_navigation

### CODE REVIEW

File: src/flowNavigation/FlowNavigation.spec.js [hunk 1]
```diff
@@ -26,11 +27,7 @@ describe('Flow navigation', () => {
   }
 
   function backButton() {
-    return component.find('.tw-flow-navigation__back-button');
-  }
-
-  function backButtonHidden() {
-    return backButton().hasClass('tw-flow-navigation__back-button--hidden');
+    return component.find(BackButton);
```
Comment (codewhisperer) [inline src/flowNavigation/FlowNavigation.spec.js]: Would it be cleaner if you added `!!` before?

Comment (Tankenstein) [reply src/flowNavigation/FlowNavigation.spec.js]: No - it's used to actually find the component and do stuff with it, not do a truthy check

File: src/flowNavigation/backButton/BackButton.js [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+
+  render() {
+    const { steps, activeStep, onGoBack } = this.props;
+
+    return (
+      <button
+        className={`btn-unstyled visible-xs tw-flow-navigation__back-button ${
+          !onGoBack ? 'tw-flow-navigation__back-button--hidden' : ''
+        }`}
+        onClick={() => onGoBack && onGoBack()}
```
Comment (codewhisperer) [inline src/flowNavigation/backButton/BackButton.js]: I don't get this one. Is the first condition checking if the function is there?

File: src/flowNavigation/FlowNavigation.spec.js (hunk 1 above)
Comment (codewhisperer) [reply src/flowNavigation/FlowNavigation.spec.js]: But shouldn't the function have a different name then?

Comment (codewhisperer) [reply src/flowNavigation/FlowNavigation.spec.js]: Oh never mind, thought the function name was backButtonHidden.

**Pull request merged**